### ISSUE TITLE
`torchgen` relies on non-packaged `native_functions.yaml`

### ISSUE THREAD

**Issue by antoniojkim** [pytorch/pytorch#77445]
### 🐛 Describe the bug

There is this assumption here
https://github.com/pytorch/pytorch/blob/6066e5929f12cdef52fe5d85af8001ad3583ec42/torchgen/gen_lazy_tensor.py#L232-L234
which is primarily there to locate these yamls
https://github.com/pytorch/pytorch/blob/6066e5929f12cdef52fe5d85af8001ad3583ec42/torchgen/gen_lazy_tensor.py#L302-L304
However, this assumption means that `torchgen` is not usable as a "pip installable" package. So, although its packaged alongside `torch`, it is unusable outside of a PyTorch clone.

This appears to be an oversight, so opening an issue to discuss ways to resolve this.

### Versions

N/A

**Comment by wconstab** [comment]
@bdhirsh do you know if this is just an oversight or is there more to this story?

**Comment by bdhirsh** [comment]
Yeah, it's an oversight (@ezyang pointed it out last week). Right now you basically need to have a local copy of `native_functions.yaml` that you can point to. We should figure out how to package it properly.

**Comment by ezyang** [comment]
@zou3519 was going to figure this out but he's on vacation, if someone wants to beat him to the punch please do haha

**Comment by antoniojkim** [comment]
@zou3519 Any updates on this?

**Comment by zou3519** [comment]
It's coming soon (in 1-2 weeks at the rate I am going). The problem I'm running into is getting it to work on windows. How urgently do you need this?

**Comment by antoniojkim** [comment]
Not super urgently. It is blocking the Torch-MLIR LTC integration being merged into the main development branch. However, we can continue working off the staging branch until that time. 1-2 weeks timeframe should be fine.

**Comment by zou3519** [comment]
@antoniojkim another question -- would you want this in PT 1.12 (in June), or is it fine if this is in the nightlies until e.g. PT1.13

**Comment by antoniojkim** [comment]
My need for it is LTC related which if my understanding is correct isn't going to be stable in 1.12. So, its fine to keep it in nightly until 1.13

**Comment by antoniojkim** [comment]
@zou3519 Any updates on this?

**Comment by zou3519** [comment]
@antoniojkim torchgen and torch are packaged together (e.g. pip install torch will install both). The native_functions.yaml that is packaged in torchgen is a copy of the one used to build that version of torch.

More concretely, we added a step in our build process that copies the native_functions.yaml in PyTorch to torchgen. So the native_functions.yaml in aten/src/ATen/native is still the source of truth, and the one in torchgen is a mirror of it. Does that help answer the question?

**Comment by antoniojkim** [comment]
ah, that makes sense. Thanks for the response!

**Comment by antoniojkim** [comment]
I think this is working now. Closing the issue.
